src/dht.cpp

```diff
@@ -575,7 +575,75 @@ struct Dht::Search {
      */
     time_point getNextStepTime(const std::map<ValueType::Id, ValueType>& types, time_point now) const;
 
-    bool removeExpiredNode(time_point now);
+    /**
+     * Removes up to `count` nodes which have been expired for at least
+     * NODE::NODE_EXPIRE_TIME minutes.
+     *
+     * @param now  The reference to now. Passing time_point::max() will remove
+     *             up to `count` expired nodes regardless of the time its been
+     *             expired.
+     * @param count  The number of expired nodes to remove.
+     *
+     * @return true if a node has been removed, else false.
+     */
+    bool removeExpiredNodes(time_point now, size_t count) {
```

Comment:
Yeah but why removing up to count nodes if a search expires ? For instance why not removing all expired nodes in that case ?